src/extractor/scripting_environment_lua.cpp

```diff
@@ -202,6 +205,20 @@ void LuaScriptingEnvironment::InitContext(LuaScriptingContext &context)
              .def(luabind::constructor<>())
              // Dear ambitious reader: registering .location() as in:
              // .def("location", +[](const osmium::NodeRef& nref){ return nref.location(); })
+             .def("location", +[](const osmium::NodeRef& nref){
+                 if(location_sparse_handler)
+                 {
+                     osmium::Location location = LuaScriptingEnvironment::location_sparse_handler->get_node_location(nref.ref());
+                     return location;
+                 }
+                 else if(location_sparse_handler)
+                 {
+                     osmium::Location location = LuaScriptingEnvironment::location_sparse_handler->get_node_location(nref.ref());
+                     return location;
+                 }
+                 else
+                     return nref.location();
```

Comment:
Yes, but what should I do when the LUA script tries to access the cache that is not generated?
I propose to catch location exception and propagate with complement information about how generate cache.